### ISSUE TITLE
Question: proxychains mount- and ping-commands and UDP-Protocols through socks5/http-Proxy

### ISSUE THREAD

**Issue by RichardBeisser** [rofl0r/proxychains-ng#336]
Hello,

is it possible to use proxychains with mount-command or ping-command?
Can I use proxychains for UDP-Protocolls like ntp for sync time, or RTP for Voip?

If yes, how?

Thanks.

**Comment by rofl0r** [comment]
udp: no
icmp (ping): no
mount webdav? depends on whether it uses TCP behind the scenes, if so it should be possible.

**Comment by RichardBeisser** [comment]
you said that proxychains do not support udp
but socks5 supports  UDP:
https://en.wikipedia.org/wiki/SOCKS#SOCKS5
Could you explain why it do not work?

when I mount a webdav-Drive using mount, tcp will be used. Do you know how I get working it with proxychains?

**Comment by rofl0r** [comment]
> but socks5 supports UDP

theoretically yes, but i've so far not seen any socks5 server software that supports it (and the 2 i authored myself don't either). most people go for the low-hanging fruit

**Comment by RichardBeisser** [comment]
here is an example of a socks5 server which supports.
Does proxychains support UDP at the moment, or do you have to change something in the software?

**Comment by rofl0r** [comment]
> Does proxychains support UDP at the moment

no

> do you have to change something in the software?

yes, we'd need to hook the libc functions that send UDP datagrams, handle the socks5 udp protocol and add special-case code to reject socks4 and http proxies for this scenario.

**Comment by RichardBeisser** [comment]
> yes, we'd need to hook the libc functions that send UDP datagrams, handle the socks5 udp protocol and add special-case code to reject socks4 and http proxies for this scenario.

have you plans to do this?
I am asking because of see usecase above.

**Comment by rofl0r** [comment]
> have you plans to do this?

no, i've never needed this functionality. i may consider to merge it though if someone comes up with a well-written PR.

**Comment by NyaMisty** [comment]
good test tool: gost

**Comment by eladts** [comment]
> i may consider to merge it though if someone comes up with a well-written PR.

Looks like @hc-syn did just that.

https://github.com/rofl0r/proxychains-ng/pull/570

https://www.synacktiv.com/sites/default/files/2025-06/sstic-2025-bbs-proxychains-udp.pdf
